pages/studentPortal/quizcomponent/FirstProgrammingLanguageQuiz.tsx

```diff
@@ -0,0 +1,182 @@
+import LangResults from "../../../components/resources/quizzes/langQuiz/LangResults";
+import BluePrint from "../../../components/resources/quizzes/shared/BluePrint";
+import SkillSelections from "../../../components/resources/quizzes/shared/SkillSelections";
+import StartQuiz from "../../../components/resources/quizzes/shared/StartQuiz";
+export enum Stage {
+  START,
+  LEARNCODING,
+  WORKFIELDS,
+  BUILDTARGET,
+  BLUEPRINT,
+  RESULTS,
+}
+
+type QuizSelection = {
+  name: string;
+  result?: string;
+  weight?: number;
+};
+
+const FirstProgrammingLanguageQuiz = () => {
+  const quiz: QuizSelection[][] = [
+    [
+      { name: "Learn an in-demand skill" },
+      { name: "Work in tech" },
+      { name: "Build programs for fun" },
+      { name: "I run a startup" },
+      { name: "Become a professional developer" },
+      { name: "Communicate better with technical coworkers" },
+      { name: "Start a side hustle" },
+    ],
+    [
+      { name: "Front end development", result: "JavaScript", weight: 1 },
+      { name: "Back end development", result: "JavaScript", weight: 1 },
+      { name: "Game development", result: "JavaScript", weight: 1 },
+      { name: "Mobile development", result: "JavaScript", weight: 1 },
+      { name: "Data analytics", result: "Python", weight: 1 },
+      { name: "Product management", result: "JavaScript", weight: 1 },
+      { name: "Digital marketing", result: "JavaScript", weight: 1 },
+      { name: "UX/UI design", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
+    ],
+    [
+      { name: "Websites", result: "JavaScript", weight: 1 },
+      { name: "Storefront", result: "HTML/CSS", weight: 1 },
+      { name: "Mobile apps", result: "JavaScript", weight: 1 },
+      { name: "Games", result: "JavaScript", weight: 1 },
+      { name: "Email", result: "HTML/CSS", weight: 1 },
+      { name: "Tools to automate your life", result: "JavaScript", weight: 1 },
+      { name: "I don't know / Not sure yet", result: "JavaScript", weight: 1 },
+    ],
+  ];
+
+  const pageOptions = quiz.map((pages) => {
+    return pages.map((options) => {
+      return options.name;
+    });
+  });
+
+  type PageContent = {
+    title: string;
+    body: string;
+  };
+
+  const quizQuestions: PageContent[] = [
+    {
+      title: "What Coding Language Should I Learn First?",
+      body: "Take this free quiz to find out what coding language you should learn first.",
+    },
+    {
+      title: "Why do you want to learn coding?",
+      body: "Select all that apply.",
+    },
+    {
+      title: "What field of work are you interested in?",
+      body: "Select 1-3 choices.",
+    },
+    {
+      title: "What are you interested in building?",
+      body: "Select all that apply.",
+    },
+  ];
+
+  const [quizResponses, setQuizResponses] = useState([]);
+  const [selectedSkill, setSelectedSkill] = useState<Array<Array<string>>>([]);
```

Comment:
what is the difference between quizResponses, and selectedSkill?

I think if you change Array<Array<string>> to a custom QuizState object that might clear things up, but I'm not sure how your two states are different

Reply:
I might be missing something, but I think you can clean up these state variables and come up with a cleaner/more readable solution

Reply:
Resolved!